src/runtime/recipe/connection-constraint.ts

```diff
@@ -17,7 +17,7 @@ import { Particle } from './particle.js';
 export abstract class EndPoint implements Comparable<EndPoint> {
   abstract _compareTo(other: EndPoint): number;
   abstract _clone(cloneMap?: CloneMap);
```

Comment:
Shouldn't clone also get a type, even if `void`?

Reply:
someday yes.  This was a quick pass through some code to pass the time on the flight from SYD...